compiler/ballerina-lang/src/main/java/io/ballerina/compiler/api/symbols/ConstantSymbol.java

```diff
@@ -22,12 +22,19 @@
  *
  * @since 2.0.0
  */
-public interface ConstantSymbol extends VariableSymbol {
+public interface ConstantSymbol extends VariableSymbol, SingletonTypeSymbol {
 
     /**
      * Get the constant value.
      *
      * @return {@link Object} value of the constant
      */
     Object constValue();
+
+    /**
+     * Gets the broader type of constant expression associated with this symbol.
+     *
+     * @return {@link TypeSymbol} of the broader type
+     */
+    TypeSymbol broaderTypeDescriptor();
```

Comment:
What if we return an Optional here?

Reply:
From a language POV, no reason for broader type of a constant expr to be optional right? If there's a constant, it also has a broader type. It may become `null` for incomplete constant declarations?

Reply:
Shouldn't incomplete constant declarations is a expected case?

Reply:
Yes, but the APIs for the most part designed to be in-line with the spec. Optionals were used in places where the spec has specified it's optional. e.g., rest param in function type